model.py

```diff
@@ -5988,22 +5982,22 @@ def best_covers_among(cls, resources):
                 # A Resource below the minimum quality threshold is not
                 # usable, period.
                 continue
-            if not champions or quality > champion_score:
+
+            # In order, our criteria are: whether we
+            # mirrored the representation (which means we directly
+            # control it), image quality, and media type suitability.
+            #
+            # We invert media type suitability because it's given to us
+            # as a priority (where smaller is better), but we want to compare
+            # it as a quantity (where larger is better).
+            compare_key = (rep.mirror_url is not None, quality, -media_priority)
+            if not champion_key or (compare_key > champion_key):
+                # A new champion.
```

Comment:
The overall change in this method is to add 'did we personally mirror this representation' as a criterion for quality. To add this I got rid of the complexity of nested if-then statements to determine whether one representation is better than another. Instead we create a 3-tuple key such that a better Representation will > a worse Representation, and then compare the keys to see which is best.